plugin/core/views.py

```diff
@@ -780,13 +780,12 @@ def text2html(content: str) -> str:
 
 
 def make_link(href: str, text: Any, class_name: Optional[str] = None, tooltip: Optional[str] = None) -> str:
-    if isinstance(text, str):
-        text = text.replace(' ', '&nbsp;')
     link = "<a href='{}'".format(href)
     if class_name:
         link += " class='{}'".format(class_name)
     if tooltip:
         link += " title='{}'".format(html.escape(tooltip))
+    text = text2html(str(text)).replace(' ', '&nbsp;')
```

Comment:
make_link  is called inside the `format_completion` 
I would say that calling text2html inside here will affect the completion performance

I will test with
https://github.com/sublimelsp/LSP-performance-test

Reply:
this branch
Best of 3: 0.2930799584137276 usec. All times: [0.2931855833157897, 0.29389379161875695, 0.2930799584137276]
Best of 3: 0.2897154582897201 usec. All times: [0.2897154582897201, 0.29021529166493565, 0.2925562916789204]

main branch
Best of 3: 0.20980629161931574 usec. All times: [0.21008566673845053, 0.20980891666840762, 0.20980629161931574]
Best of 3: 0.20813845819793642 usec. All times: [0.20813845819793642, 0.20843349990900606, 0.20843862497713417]

Reply:
Yes, it will affect performance because it's an extra logic. But it's a logic that has to run if we want things to work correctly in all circumstances.

That said, I will look into it a bit more to see if we can do "fast paths" in some cases.
But I don't want to decide outside of such functions whether to escape or not because that would make the logic error-prone in regards to future changes.

Applies to all 3 of your comments.

Reply:
All of my comments have one thing in common, to call the text2html less times.

> That said, I will look into it a bit more to see if we can do "fast paths" in some cases.

+1

Reply:
Pushed this code 964e5a8
It makes the format completion function faster